src/main/java/donnafin/logic/commands/AppendCommand.java

```diff
@@ -54,12 +72,26 @@
      * @param policy the new policy being added.
      */
     public AppendCommand(PersonAdapter personAdapter, Policy policy) {
+        requireNonNull(personAdapter);
+        requireNonNull(policy);
         this.personAdapter = personAdapter;
         this.hashableNewValue = policy;
         this.editor = pa -> {
             Set<Policy> policies = new HashSet<>(pa.getSubject().getPolicies());
-            policies.add(policy);
-            pa.editPolicies(policies);
+            boolean containsSimilarName = policies.stream().anyMatch(item ->
+                    item.getName().equalsIgnoreCase(policy.getName()));
+            if (policies.contains(policy)) {
+                try {
+                    messageRESULT = MESSAGE_DUPLICATE_POLICY;
+                    throw new CommandException(MESSAGE_DUPLICATE_POLICY);
+                } catch (CommandException e) {
+                    e.printStackTrace();
+                }
```

Comment:
Hmm, why are we throwing a command exception only to swallow it with a printstactrace? I don't quite get the code here.

Reply:
Hmmm I'm not sure if pa (the lambda function) can throw an exception cause when I delete the try catch block it'll just show up as an error

Reply:
Oh then why the throw new CE at all?

Reply:
alright, I deleted the CE!